src/main/java/com/google/sps/servlets/SeedDatabaseServlet.java

```diff
@@ -0,0 +1,89 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.Entity;
+import com.google.common.base.Charsets;
+import com.google.common.io.Resources;
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.sps.data.DatastoreAccess;
+import com.google.sps.data.Mentee;
+import com.google.sps.data.Mentor;
+import com.google.sps.data.UserAccount;
+import com.google.sps.util.DummyDataConstants;
+import com.google.sps.util.ErrorMessages;
+import com.google.sps.util.ParameterConstants;
+import com.google.sps.util.ResourceConstants;
+import com.google.sps.util.URLPatterns;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.stream.Collectors;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Seeds the database with based on a large input dataset in JSON format. Upon calling HTTP GET,
+ * this servlet attempts to seed the database and returns a JSON object representing the operations
+ * success.
+ *
+ * @author sylviaziyuz
+ * @author guptamudit
+ * @version 1.0
+ * @param URLPatterns.SEED_DB this servlet serves requests at /seed-db
+ */
+@WebServlet(URLPatterns.SEED_DB)
+public class SeedDatabaseServlet extends HttpServlet {
```

Comment:
This is pretty odd to have. In general the seeding shouldn't expose an endpoint that a malicious entity can take advantage of (should be done as part of the setup based on config indicating whether seeding is required) but leaving it as ok for now. Let's be sure to clean it up later.